logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java

```diff
@@ -0,0 +1,857 @@
+        if (queryResult.hasError())
+        {
+            Assert.fail("Error executing query [" +  query.getCommand() + "] : " + queryResult.getError());
+        }
+
+        // Create database
+
+        query = new Query("CREATE DATABASE " + TEST_DATABASE);
+        queryResult = influxDB.query(query);
+        if (queryResult.hasError())
+        {
+            Assert.fail("Error executing query [" +  query.getCommand() + "] : " + queryResult.getError());
+        }
+        echo("InfluxDB test database cleared and prepared");
+    }
+
+    @Test
+    @UseDataProvider("testBulkPutProvider")
+    public void testBulkPut(List<Map<Field, InfluxDBType>> insertedAndExpectedPoints, String measurement,
+                            String timeField, TimeUnit format, CONFIG_MODE configMode, Set<String> explicitTags,
+                            Set<String> explicitFields)
+            throws InitializationException {
+
+        final TestRunner runner = TestRunners.newTestRunner("com.hurence.logisland.processor.datastore.BulkPut");
+
+        final InfluxDBControllerService service = new InfluxDBControllerService();
+        runner.addControllerService("influxdb_service", service);
+        runner.setProperty(service, InfluxDBControllerService.URL.getName(), INFLUXDB_URL);
+        runner.setProperty(service, InfluxDBControllerService.DATABASE.getName(), TEST_DATABASE);
+        runner.setProperty(service, InfluxDBControllerService.MODE.getName(), configMode.toString());
+        /**
+         * NOTE: we always use explicit time field declaration as using record_time (which is the default if no time field
+         * is specified with a measurement) will generate a lot of records with the same record_time which ends up with
+         * point being overwritten in InfluxDB. To prevent that, we force usage of a time field that we use in different
+         * formats for testing purpose.
+         */
+        String explicitTagsConfigString = makeTagsConfigString(measurement, configMode, explicitTags);
+        if (explicitTagsConfigString != null)
+        {
+            runner.setProperty(service, InfluxDBControllerService.TAGS.getName(), explicitTagsConfigString);
+        }
+        String explicitFieldsConfigString = makeFieldsConfigString(measurement, configMode, explicitFields);
+        if (explicitFieldsConfigString != null)
+        {
+            runner.setProperty(service, InfluxDBControllerService.FIELDS.getName(), explicitFieldsConfigString);
+        }
+        runner.setProperty(service, InfluxDBControllerService.TIME_FIELD.getName(),
+                makeTimeFieldConfigString(measurement, timeField, format));
+        runner.setProperty(service, InfluxDBControllerService.FLUSH_INTERVAL.getName(), "1000");
+        runner.setProperty(service, InfluxDBControllerService.BATCH_SIZE.getName(), "500");
+        runner.enableControllerService(service);
+
+        runner.assertNotValid();
```

Comment:
runner.assertNotValid(service) instead assertNotValid() test only validation of the processor

Reply:
Done